src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,63 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private LinkedList<Entry<K, V>>[] buckets;
+    private int size;
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public MyHashMap() {
+        buckets = new LinkedList[INITIAL_CAPACITY];
+        size = 0;
+    }
+
+    private int getBucketIndex(K key) {
+        return Math.abs(key.hashCode()) % buckets.length;
+    }
```

Comment:
The `getBucketIndex` method should have an access modifier. Consider making it private as it is a utility method used only within this class.